src/sentry/mediators/external_requests/util.py

```diff
@@ -56,21 +56,21 @@ def send_and_save_sentry_app_request(url, sentry_app, org_id, event, **kwargs):
 
     try:
         resp = safe_urlopen(url=url, **kwargs)
-    except RequestException:
+        track_response_code(resp.status_code, slug, event)
```

Comment:
@NisanthanNanthakumar I think putting everything here before the except block which is only catching a timeout error that happens on like 58 is hard to read. Why not:
```
    try:
        resp = safe_urlopen(url=url, **kwargs)
   except Timeout:
      # timeout stuff
   finally:
     # normal stuff
     resp.raise_for_status()
     return resp
```